tensorflow/c/c_api.cc

```diff
@@ -2803,4 +2803,43 @@ TF_Buffer* TF_GetRegisteredKernelsForOp(const char* name, TF_Status* status) {
   }
   return ret;
 }
+
+// TF_Server functions ----------------------------------------------
+
+TF_Server::TF_Server(tensorflow::ServerInterface* server) : server(server) {}
+
+TF_Server* TF_NewServer(const void* proto, size_t proto_len,
+                        TF_Status* status) {
+  tensorflow::ServerDef server_def;
+  if (!server_def.ParseFromArray(proto, static_cast<int>(proto_len))) {
+    status->status = InvalidArgument("Unparseable ServerDef");
+    return nullptr;
+  }
+
+  auto out_server = new std::unique_ptr<tensorflow::ServerInterface>();
```

Comment:
Isn't `out_server` being leaked here since the `std::unique_ptr` is being heap allocated?
See comment in `c_api_internal.h`, I think we can simplify this to:

```c++
std::unique_ptr<TF_Server> ret(new TF_Server);
status->status = tensorflow::NewServer(server_def, &ret->server);
if (!status->status.ok()) return nullptr;
return ret.release();
```